base/common/EnumReflection.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <type_traits>
+#include <fmt/format.h>
+#include <../../contrib/magic_enum/include/magic_enum.hpp> //FIXME
+
+// Can't name concept is_enum as is would conflict with type trait
+template <class T> concept type_is_enum = std::is_enum_v<T>;
```

Comment:
May be the first PR containing `concept`)

Reply:
Ok if it's supported by (both) our compilers...

Reply:
There were issues with gcc and some google contrib, I'll fix them if they arise again.